Markdown/Symbol List - Heading.tmPreferences

```diff
@@ -9,9 +9,14 @@
 		<integer>1</integer>
 		<key>symbolTransformation</key>
 		<string><![CDATA[
-		s/\s*#*\s*\z//g;         # strip trailing space and #'s
-		s/(?<=#)#/ /g;           # change all but first # to m-space
-		s/^#( *)\s+(.*)/$1$2/;   # strip first # and space before title
+			s/\[([^]]+)\]\([^)]+\)/$1/g; 		# beautify links
+			s/\[([^]]+)\](?:\[[^]]+\])?/$1/g; 	# beautify references
+			s/^\s*//g;							# strip leading whitespace
+			s/\s*#+\s*\z//g;					# strip trailing hashes
+			s/(?<=#)#/ /g;						# change all but first # to m-space
+			s/^#( *)\s+(.*)/$1$2/;				# strip first # and space before title
```

Comment:
can you explain this one a bit please? It looks like it could cause catastrophic backtracking because it matches space star followed by any whitespace plus followed by anything star...
It takes the text where all the leading hashes except the first one were turned into spaces, and then I'm a bit lost. The first capture group will take all but the last space before the title text starts, so if the heading was `## Hello`, then it will take the space which was converted from the second hash. And it is kept in the output, so is the comment correct? Or am I missing something?

Reply:
Thes are the lines, which existed in this file before, so I kept them in place. If it is suspicious it was so before this PR, too. We may need to forward your question to the original author - or just replace it by the more explicit solution from  MarkdownEditing  3.0.0 which is basically the same as the one from @jfcherng's gist.

Reply:
Apologies, I didn't notice it was there before. Thanks for the improvements!

Reply:
Not at all. Existing before or not. If it's suspicious we need to at least think about it. We don't need to keep possible trouble makers.